chia/wallet/trading/trade_store.py

```diff
@@ -115,6 +165,15 @@ async def add_trade_record(self, record: TradeRecord) -> None:
                 ),
             )
             await cursor.close()
+            # remove all current coin ids
+            await conn.execute("DELETE FROM coin_of_interest_to_trade_record WHERE trade_id=?", (record.trade_id,))
```

Comment:
here, you don't turn `trade_id` into hex (which is correct). But I would expect there to be a test catching this error, since you added the entry as hex in the migration function. I think you should add one that ensures entries are correctly removed.

Perhaps it's the migration function that doesn't have test coverage.

Reply:
true, migration function was off. I added migration and trade record replacement tests.